modular_skyrat/modules/reagent_forging/code/mindless_killer.dm

```diff
@@ -0,0 +1,47 @@
+/datum/component/mindless_killer
+	///how much force is added to items before hitting a mindless mob
+	var/mindless_force = 10
+	///how much force is multiplied to items before hitting a mindless mob (applied after mindless force)
+	var/mindless_multiplier = 1
+	///the obj parent this component is attached to
+	var/obj/obj_parent
+
+/datum/component/mindless_killer/Initialize(mindless_force_override = 10, mindless_multiplier_override = 1)
+	if(!isobj(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	obj_parent = parent
+
+	mindless_force = mindless_force_override
+	mindless_multiplier = mindless_multiplier_override
+
+	RegisterSignal(parent, COMSIG_ATOM_EXAMINE, PROC_REF(examine))
+	RegisterSignal(parent, COMSIG_ITEM_PRE_ATTACK, PROC_REF(do_force))
+	RegisterSignal(parent, COMSIG_ITEM_AFTERATTACK, PROC_REF(undo_force))
```

Comment:
Butchering interrupts and cancels the attack chain in the on_attack point. These bonuses are applied in pre_attack, and removed in post_attack. The chain goes `pre_attack -> on_attack -> attack -> post_attack` Meaning you can stack these bonuses by butchering and cancelling the action repeatedly.